framework/contracts/account/manager/src/commands.rs

```diff
@@ -735,6 +781,56 @@ pub fn update_internal_config(deps: DepsMut, info: MessageInfo, config: Binary)
     }
 }
 
+fn query_ownership(deps: Deps, maybe_manager: Addr) -> ManagerResult<String> {
+    let Ownership::<String> { owner, .. } = deps
+        .querier
+        .query_wasm_smart(maybe_manager, &QueryMsg::Ownership {})?;
+
+    owner.ok_or(ManagerError::Admin(cw_controllers::AdminError::NotAdmin {}))
+}
+
+fn assert_admin_right(deps: Deps, sender: &Addr) -> ManagerResult<()> {
+    let ownership_test = cw_ownable::assert_owner(deps.storage, sender);
+
+    if ownership_test.is_ok() {
+        return Ok(());
+    }
+
+    let account_info = INFO.load(deps.storage)?;
+    match account_info.governance_details {
+        // If the account has SubAccount governance, the owner of the manager also has admin rights over this account
+        GovernanceDetails::SubAccount { manager } => {
+            // We try to query the ownership of the manager monarch account if the first query failed
+            let mut current = manager;
+            let mut i = 0;
+            loop {
+                if let Ok(owner) = query_ownership(deps, current) {
+                    // If the owner account is indeed owned by another instance
+                    if *sender == owner {
+                        // If the owner of the current contract is the sender, the admin test is passed
+                        return Ok(());
+                    } else {
+                        // If not, we try again with the queried owner
+                        current = deps.api.addr_validate(&owner)?
+                    }
+                } else {
+                    return Err(ManagerError::Admin(cw_controllers::AdminError::NotAdmin {}));
+                }
+                if {
+                    i += 1;
+                    i
+                } > MAX_ADMIN_RECURSION
```

Comment:
I would prefer for it to even not possible to create >5 level deep sub-account, we can save "depth" in manager config

Reply:
I'm not sure that's the way to go.
Because we are using the cw-ownable library for ownership, we allow for ownership of accounts indirectly and this is not followed inside abstract. I don't think enforcing subaccount maximum depth helps us here.
And we are not limiting the depth at runtime, only the depth at which you can call a subaccount action.
WDYT ?